java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map

```
ERROR [pool-7-thread-1] o.z.s.f.PluginEventsProcessor - java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203) ~[guava-18.0.jar!/:?]
at com.google.common.cache.LocalCache.get(LocalCache.java:3937) ~[guava-18.0.jar!/:?]
at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941) ~[guava-18.0.jar!/:?]
at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824) ~[guava-18.0.jar!/:?]
at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4830) ~[guava-18.0.jar!/:?]
at org.zalando.stups.fullstop.plugin.provider.impl.TaupageYamlProviderImpl.apply(TaupageYamlProviderImpl.java:83) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.plugin.provider.impl.TaupageYamlProviderImpl.apply(TaupageYamlProviderImpl.java:27) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.plugin.impl.EC2InstanceContextImpl.getTaupageYaml(EC2InstanceContextImpl.java:183) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.plugin.taupageyaml.TaupageYamlPlugin.process(TaupageYamlPlugin.java:35) ~[fullstop-taupage-yaml-plugin-1.2.0-SNAPSHOT.jar!/:?]
at java.util.ArrayList.forEach(ArrayList.java:1249) ~[?:1.8.0_66-internal]
at org.zalando.stups.fullstop.plugin.AbstractEC2InstancePlugin.processEvent(AbstractEC2InstancePlugin.java:36) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.PluginEventsProcessor.doProcess(PluginEventsProcessor.java:52) ~[fullstop-processing-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.PluginEventsProcessor.doProcess(PluginEventsProcessor.java:43) ~[fullstop-processing-1.2.0-SNAPSHOT.jar!/:?]
at java.util.ArrayList.forEach(ArrayList.java:1249) [?:1.8.0_66-internal]
at org.zalando.stups.fullstop.PluginEventsProcessor.process(PluginEventsProcessor.java:33) [fullstop-processing-1.2.0-SNAPSHOT.jar!/:?]
at com.amazonaws.services.cloudtrail.processinglibrary.reader.EventReader.emitEvents(EventReader.java:229) [aws-cloudtrail-processing-library-1.0.2.jar!/:?]
at com.amazonaws.services.cloudtrail.processinglibrary.reader.EventReader.processSource(EventReader.java:156) [aws-cloudtrail-processing-library-1.0.2.jar!/:?]
at com.amazonaws.services.cloudtrail.processinglibrary.AWSCloudTrailProcessingExecutor$ScheduledJob$1.run(AWSCloudTrailProcessingExecutor.java:178) [aws-cloudtrail-processing-library-1.0.2.jar!/:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_66-internal]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_66-internal]
at java.lang.Thread.run(Thread.java:745) [?:1.8.0_66-internal]
Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
at org.zalando.stups.fullstop.plugin.provider.impl.TaupageYamlProviderImpl.lambda$getTaupageYaml$9(TaupageYamlProviderImpl.java:65) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_66-internal]
at org.zalando.stups.fullstop.plugin.provider.impl.TaupageYamlProviderImpl.getTaupageYaml(TaupageYamlProviderImpl.java:65) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
at org.zalando.stups.fullstop.plugin.provider.impl.TaupageYamlProviderImpl.access$000(TaupageYamlProviderImpl.java:27) ~[fullstop-instance-plugin-support-1.2.0-SNAPSHOT.jar!/:?]
```

--- a/fullstop-aws-support/fullstop-aws-userdata-support/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java
+++ b/fullstop-aws-support/fullstop-aws-userdata-support/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java
@@ -51,8 +51,8 @@ public class UserDataProvider {
         byte[] bytesUserData = Base64.decode(userData);
         String decodedUserData = new String(bytesUserData);
 
-        Yaml yaml = new Yaml();
+        final Object yamlData = new Yaml().load(decodedUserData);
 
-        return (Map) yaml.load(decodedUserData);
+        return yamlData instanceof Map ? (Map) yamlData : null;
     }
 }

--- a/fullstop-plugins/fullstop-instance-plugin-support/src/main/java/org/zalando/stups/fullstop/plugin/provider/impl/TaupageYamlProviderImpl.java
+++ b/fullstop-plugins/fullstop-instance-plugin-support/src/main/java/org/zalando/stups/fullstop/plugin/provider/impl/TaupageYamlProviderImpl.java
@@ -57,12 +57,16 @@ public class TaupageYamlProviderImpl implements TaupageYamlProvider {
                                         .withInstanceId(instanceId)
                                         .withAttribute(USER_DATA));
 
+                final Yaml yaml = new Yaml();
+
                 return ofNullable(response)
                         .map(DescribeInstanceAttributeResult::getInstanceAttribute)
                         .map(InstanceAttribute::getUserData)
                         .map(Base64::decode)
                         .map(String::new)
-                        .map(data -> (Map) new Yaml().load(data));
+                        .map(yaml::load)
+                        .filter(data -> data instanceof Map) // everything else is obviously no valid taupage format
+                        .map(data -> (Map) data);
 
             } catch (AmazonClientException e) {
                 log.warn("Could not get Taupage YAML for instance: " + instanceId, e);

--- a/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/TaupageYamlProviderImplTest.java
+++ b/fullstop-plugins/fullstop-instance-plugin-support/src/test/java/org/zalando/stups/fullstop/plugin/provider/impl/TaupageYamlProviderImplTest.java
@@ -62,6 +62,30 @@ public class TaupageYamlProviderImplTest {
     }
 
     @Test
+    public void testApplyWithTaupageAmiButInvalidYaml() throws Exception {
+        // a yaml list is not a valid taupage format. Map is required.
+        final String yamlData = "- a\n- b\n- c\n";
+
+        when(ec2InstanceContextMock.isTaupageAmi()).thenReturn(Optional.of(true));
+
+        when(ec2InstanceContextMock.getInstanceId()).thenReturn(INSTANCE_ID);
+        when(ec2InstanceContextMock.getClient(eq(AmazonEC2Client.class))).thenReturn(amazonEC2ClientMock);
+        when(amazonEC2ClientMock.describeInstanceAttribute(any())).thenReturn(new DescribeInstanceAttributeResult().
+                withInstanceAttribute(new InstanceAttribute()
+                        .withUserData(Base64.encodeAsString(yamlData.getBytes()))));
+
+        Optional<Map> result = taupageYamlProvider.apply(ec2InstanceContextMock);
+
+        assertThat(result).isEmpty();
+
+
+        verify(ec2InstanceContextMock).isTaupageAmi();
+        verify(ec2InstanceContextMock).getInstanceId();
+        verify(ec2InstanceContextMock).getClient(eq(AmazonEC2Client.class));
+        verify(amazonEC2ClientMock).describeInstanceAttribute(any());
+    }
+
+    @Test
     public void testApplyWithOtherAmi() throws Exception {
         when(ec2InstanceContextMock.isTaupageAmi()).thenReturn(Optional.of(false));
 

--- a/fullstop-violation/fullstop-violation-jpa/src/main/java/org/zalando/stups/fullstop/violation/service/impl/ApplicationLifecycleServiceImpl.java
+++ b/fullstop-violation/fullstop-violation-jpa/src/main/java/org/zalando/stups/fullstop/violation/service/impl/ApplicationLifecycleServiceImpl.java
@@ -108,6 +108,7 @@ public class ApplicationLifecycleServiceImpl implements ApplicationLifecycleServ
                 .map(base64Decoder::decode)
                 .map(String::new)
                 .map(yaml::load)
+                .filter(data -> data instanceof Map)
                 .map(map -> (Map) map);
 
         final Optional<ApplicationEntity> application = taupageYaml

